src/sentry/integrations/opsgenie/integration.py

```diff
@@ -63,8 +128,42 @@ class OpsgenieIntegrationProvider(IntegrationProvider):
     features = frozenset([IntegrationFeatures.INCIDENT_MANAGEMENT, IntegrationFeatures.ALERT_RULE])
     requires_feature_flag = True  # limited release
 
+    def get_account_info(self, base_url, api_key):
+        client = OpsgenieProxySetupClient(base_url=base_url, api_key=api_key)
+        try:
+            resp = client.get_account()
+            return resp.json
+        except ApiError as api_error:
+            logger.info(
+                "opsgenie.installation.get-account-info-failure",
+                extra={
+                    "base_url": base_url,
+                    "error_message": str(api_error),
+                    "error_status": api_error.code,
+                },
+            )
+            raise IntegrationError("The requested Opsgenie account could not be found.")
+        except (ValueError, MissingSchema) as url_error:
+            logger.info(
+                "opsgenie.installation.get-account-info-failure",
+                extra={
+                    "base_url": base_url,
+                    "error_message": str(url_error),
+                },
+            )
+            raise IntegrationError("Invalid URL provided.")
+
     def get_pipeline_views(self) -> Sequence[PipelineView]:
-        return super().get_pipeline_views()
+        return [InstallationGuideView(), InstallationConfigView()]
 
     def build_integration(self, state: Mapping[str, Any]) -> Mapping[str, Any]:
-        return super().build_integration(state)
+        api_key = state["installation_data"]["api_key"]
+        base_url = state["installation_data"]["base_url"]
+
+        account = self.get_account_info(base_url=base_url, api_key=api_key).get("data")
+
+        return {
+            "name": account.get("name"),
+            "external_id": account.get("name"),
+            "metadata": {"api_key": api_key, "base_url": base_url},
+        }
```

Comment:
I believe you'll need to add `"provider": self.key` so that the integration row can save properly.

Reply:
Actually, nvm -- looks like it picks up the provider from `self.key` 👍